packages/sourcecred/src/core/clearcred/scoredContribution.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+import type {Contribution, Expression, WeightOperand} from "./contribution";
+import {
+  type Operator,
+  applyOperator,
+  OPERATOR_KEY_PREFIX,
+  OPERATORS,
+} from "./operator";
+import {
+  getWeight,
+  hasExplicitWeight,
+  getShares,
+  getOperator,
+  type Config,
+} from "./config";
+import type {TimestampMs} from "../../util/timestamp";
+import findLast from "lodash.findlast";
+
+// TODO finish docstrings
+
+export type ScoredWeightOperand = {|
+  +key: string,
+  +subkey: string,
+  +score: number,
+|};
+
+export type ScoredExpression = {|
+  +operator: Operator,
+  +operatorKey?: string,
+  +description: string,
+  +weightOperands: $ReadOnlyArray<ScoredWeightOperand>,
+  +expressionOperands: $ReadOnlyArray<ScoredExpression>,
+  +score: number,
+|};
+
+export type ScoredContribution = {|
+  +id: string,
+  +plugin: string,
+  +type: string,
+  +timestampMs: TimestampMs,
+  +expression: ScoredExpression,
+  +participants: $ReadOnlyArray<{|
+    +id: string,
+    +shares: $ReadOnlyArray<{|
+      +amount: number,
+      +key: string,
+    |}>,
+    +score: number,
+  |}>,
+|};
+
+const scoreExpression: (Expression, Config) => ScoredExpression = (
+  expression,
+  config
+) => {
```

Comment:
why not this
```ts
(
  expression: Expression,
  config: Config
): ScoredExpression
```

this looks weird to me tbh

Reply:
accepted